libmamba/src/core/shell_init.cpp

```diff
@@ -663,7 +835,63 @@ namespace mamba
             throw std::runtime_error("Support for other shells not yet implemented.");
         }
 #ifdef _WIN32
-        enable_long_paths_support(false);
+        set_long_paths_support(1, false);
+#endif
+    }
+
+    void deinit_shell(const std::string& shell, const fs::path& conda_prefix)
+    {
+        auto mamba_exe = get_self_exe_path();
+        fs::path home = env::home_directory();
+        if (shell == "bash")
+        {
+            fs::path bashrc_path = (on_mac || on_win) ? home / ".bash_profile" : home / ".bashrc";
+            reset_rc_file(bashrc_path, conda_prefix, shell, mamba_exe);
+        }
+        else if (shell == "zsh")
+        {
+            fs::path zshrc_path = home / ".zshrc";
+            reset_rc_file(zshrc_path, conda_prefix, shell, mamba_exe);
+        }
+        else if (shell == "fish")
+        {
+            fs::path fishrc_path = home / ".config" / "fish" / "config.fish";
+            reset_rc_file(fishrc_path, conda_prefix, shell, mamba_exe);
+        }
+        else if (shell == "cmd.exe")
+        {
+#ifndef _WIN32
+            throw std::runtime_error("CMD.EXE can only be deinitialized on Windows.");
+#else
+            init_cmd_exe_registry(L"Software\\Microsoft\\Command Processor", conda_prefix, true);
+#endif
+        }
+        else if (shell == "powershell")
+        {
+            std::set<std::string> pwsh_profiles;
+            for (auto& exe : std::vector<std::string>{ "powershell", "pwsh", "pwsh-preview" })
+            {
+                auto profile_path = find_powershell_paths(exe);
+                if (!profile_path.empty())
+                {
+                    Console::stream()
+                        << "Deinit " << exe << " profile at '" << profile_path << "'";
+                    init_powershell(profile_path, conda_prefix, true);
+
+                }
+            }
+        }
+        else
+        {
+            throw std::runtime_error("Support for other shells not yet implemented.");
+        }
+
+        deinit_root_prefix(shell, conda_prefix);
+
+#ifdef _WIN32
+        if (Console::prompt("Do you want to disable long path support?", 'n')) {
+            set_long_paths_support(0, false);
```

Comment:
This is tough - I think a lot of people aren't aware of this registry setting and we do not know at all if mamba was setting it or if it was previously configured. I also think it doesn't do any harm to have it enabled. I am not sure if we should even ask to remove it...

Reply:
I guess it doesn't do any harm when leaving it enabled... Let's leave this line out then.